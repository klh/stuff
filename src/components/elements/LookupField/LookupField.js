import React from 'react';
import PropTypes from 'prop-types';
import './LookupField.css';

/**
 * LookupField could easily as well just be a functional component - did it as a component to have a few more data points for you.
 *
 */

class LookupField extends React.PureComponent {

    //PropTypes
    static propTypes = {
        submitHandler: PropTypes.func.isRequired,
        children: PropTypes.node,
        defaultValue: PropTypes.string
    };

    static defaultProps = {
        children: null,
        defaultValue: ''
    };

    render() {

        const {children, submitHandler, defaultValue} = this.props;

        return (
            <form className="form-inline" action="http://localhost:3000" method="get" onSubmit={submitHandler}>
                {children}
                <div className="form-group">
                    <input type="text" className="form-control" name="city" placeholder="City"
                           defaultValue={defaultValue}/>
                </div>
                <input type="submit" className="btn btn-default"/>
            </form>
        )
    }
}

export default LookupField;
