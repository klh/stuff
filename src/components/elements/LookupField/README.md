# LookupField readme

### Description of methods etc.

lorem ipsum dolor sti amet.

### Code examples of useage

```
 <LookupField />

```
