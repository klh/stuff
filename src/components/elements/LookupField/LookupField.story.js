import React from 'react';
import {storiesOf} from '@storybook/react';
import {action} from '@storybook/addon-actions';
import withDocs from 'storybook-readme/with-docs';
import LookupFieldReadme from './README.md';
import LookupField from './';

storiesOf('LookupField', module)
    .add('Primary',
        withDocs(LookupFieldReadme,
            () => <LookupField type="primary" onClick={action('clicked')} submitHandler={() => {
            }}>
                <span>some text to render children</span><br/>
            </LookupField>
        ));
