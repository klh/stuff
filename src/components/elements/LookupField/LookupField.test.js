import React from 'react';
import {render} from 'enzyme';

import LookupField from './';

describe('<LookupField />', () => {

    it('Snapshot matches', () => {
        const ButtonRender = render(<LookupField submitHandler={() => {
        }}/>);
        expect(ButtonRender).toMatchSnapshot();
    });
});

