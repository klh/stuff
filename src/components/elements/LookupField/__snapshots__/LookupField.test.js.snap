// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LookupField /> Snapshot matches 1`] = `
<form
  action="http://localhost:3000"
  class="form-inline"
  method="get"
>
  <div
    class="form-group"
  >
    <input
      class="form-control"
      name="city"
      placeholder="City"
      type="text"
      value=""
    />
  </div>
  <input
    class="btn btn-default"
    type="submit"
  />
</form>
`;
